### PR TITLE
port toranj test12_multi_hop_traffic

### PR DESCRIPTION
**Pull request by priyankaCh22**

### CODE REVIEW

File: silk/tests/openthread/ot_test_multi_hop_traffic.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            (self.sed_children[-1], self.sed_children[0], AddressType.MLA, AddressType.MLA),
+            (self.fed_children[0], self.fed_children[-1], AddressType.MLA, AddressType.MLA),
+        ]
+
+        timeout = 5
+        delay = 1
+        for i, (src, dst, src_address, dst_address) in enumerate(addresses):
+            port = random.randint(10000 + i * 100, 10099 + i * 100)
+            for msg_length in MSG_LENS:
+                message = random_string(msg_length)
```
Comment (tzm41) [inline silk/tests/openthread/ot_test_multi_hop_traffic.py:194]: Nice, actually good use of `random_string` function.

File: silk/tests/openthread/ot_test_multi_hop_traffic.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                src_address = src.ip6_mla
+                dst_address = dst.ip6_mla
+
+                dst.receive_udp_data(port, message, timeout)
+                time.sleep(delay)
+                src.send_udp_data(dst_address, port, message, src_address)
+
+                time.sleep(timeout - delay)
```
Comment (tzm41) [inline silk/tests/openthread/ot_test_multi_hop_traffic.py:202]: Does it take longer to send larger message?

File: silk/tests/openthread/ot_test_multi_hop_traffic.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+import random
+import time
+import unittest
+
+from silk.config import wpan_constants as wpan
+from silk.node.wpan_node import WpanCredentials
+from silk.utils import process_cleanup
+from silk.tools import wpan_table_parser
+from silk.tools.wpan_util import verify, verify_within
+from silk.unit_tests.test_utils import random_string
```
Comment (tzm41) [inline silk/tests/openthread/ot_test_multi_hop_traffic.py:29]: Import not in sorted order.

File: silk/tests/openthread/ot_test_multi_hop_traffic.py (hunk 2 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_multi_hop_traffic.py:202]: No larger messages are also received in 1 minute

File: silk/tests/openthread/ot_test_multi_hop_traffic.py (hunk 3 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_multi_hop_traffic.py:29]: Ack

**Review by tzm41 — APPROVED** [review]

File: silk/tests/openthread/ot_test_multi_hop_traffic.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+    @testcase.teardown_decorator
+    def tearDown(self):
+        pass
+
+    @testcase.test_method_decorator
+    def test01_Pairing(self):
+        # Allowlist routers with their corresponding sleepy children
+        for index in range(0, NUM_ROUTERS):
+            self.routers[index].allowlist_node(self.sed_children[index])
```
Comment (qilinkang) [inline silk/tests/openthread/ot_test_multi_hop_traffic.py:117]: remove '0, ' ?

Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_multi_hop_traffic.py:117]: Ack

**Review by qilinkang — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**